Model/Service/InvoiceService.php

```diff
@@ -48,19 +17,14 @@ public function __construct(
      */
     public function prepareInvoiceWithoutItems(OrderInterface $order, $amount)
     {
-        try {
-            $invoice = $this->orderConverter->toInvoice($order);
-            $invoice->setBaseGrandTotal($amount);
-            $invoice->setSubtotal($amount);
-            $invoice->setBaseSubtotal($amount);
-            $invoice->setGrandTotal($amount);
+        $invoice = $this->orderConverter->toInvoice($order);
+        $invoice->setBaseGrandTotal($amount);
+        $invoice->setSubtotal($amount);
+        $invoice->setBaseSubtotal($amount);
+        $invoice->setGrandTotal($amount);
 
-            $order->getInvoiceCollection()->addItem($invoice);
+        $order->getInvoiceCollection()->addItem($invoice);
```

Comment:
i know this is old code but still:

it seems funny that a method called prepareInvoiceWithoutItems pushes to the DB? (i assume thats what this is). is this a converntion based on what prepareInvoice does?

Reply:
@roopakv It is, based on original Magento method
\Magento\Sales\Model\Service\InvoiceService::prepareInvoice
```php
public function prepareInvoice(Order $order, array $qtys = [])
{
    $invoice = $this->orderConverter->toInvoice($order);
    $totalQty = 0;
    $qtys = $this->prepareItemsQty($order, $qtys);
    foreach ($order->getAllItems() as $orderItem) {
        if (!$this->_canInvoiceItem($orderItem)) {
            continue;
        }
        $item = $this->orderConverter->itemToInvoiceItem($orderItem);
        if ($orderItem->isDummy()) {
            $qty = $orderItem->getQtyOrdered() ? $orderItem->getQtyOrdered() : 1;
        } elseif (isset($qtys[$orderItem->getId()])) {
            $qty = (double) $qtys[$orderItem->getId()];
        } elseif (empty($qtys)) {
            $qty = $orderItem->getQtyToInvoice();
        } else {
            $qty = 0;
        }
        $totalQty += $qty;
        $this->setInvoiceItemQuantity($item, $qty);
        $invoice->addItem($item);
    }
    $invoice->setTotalQty($totalQty);
    $invoice->collectTotals();
    $order->getInvoiceCollection()->addItem($invoice);
    return $invoice;
 }
```